### PR TITLE
[Migrate] to null safety and [add] doc comments

### PR DESCRIPTION
**Pull request by jhb-dev**
I migrated the package to null safety and added doc comments. 

Please have a look at it and tell me if everything is okay or if I should change something :)

### CODE REVIEW

File: lib/container_tab_indicator.dart [hunk 1]
```diff
@@ -35,63 +65,63 @@ class ContainerTabIndicator extends Decoration {
...
+
   @override
   void paint(Canvas canvas, Offset offset, ImageConfiguration configuration) {
     double width;
     double height;
 
-    if (this.indicator.widthFraction != null) {
-      width = (this.indicator.widthFraction * configuration.size.width)
-          .clamp(0, configuration.size.width);
+    assert(configuration.size != null); // todo ?!
```
Comment (jhb-dev) [inline lib/container_tab_indicator.dart:83]: I wasn't exactly sure what to do with all those `configuration.size` which are nullable. I added an assert and used the bang operator (which I know is not ideal). What do you think?

File: lib/container_tab_indicator.dart [hunk 2]
```diff
@@ -2,31 +2,61 @@ library container_tab_indicator;
...
+    this.padding = const EdgeInsets.all(0.0),
+    this.color = const Color(0xFF2196F3),
+    this.radius = BorderRadius.zero,
+    this.borderWidth = 0.0,
+    this.borderColor = const Color(0xFF2196F3),
   });
 
+  /// The width of the tab indicator.
+  ///
+  /// If not provided, it uses the all the available width.
```
Comment (slovnicki) [inline lib/container_tab_indicator.dart]: typo: _the_ all the

File: lib/container_tab_indicator.dart [hunk 3]
```diff
@@ -2,31 +2,61 @@ library container_tab_indicator;
...
+
+  /// Multiplies the available width,
+  /// e.g. if set to 0.5 it uses half of the available width.
+  ///
+  /// If this is set, [width] will be ignored.
+  final double? widthFraction;
+
+  /// The height of the tab indicator.
+  ///
+  /// If not provided, it uses the all the available height.
```
Comment (slovnicki) [inline lib/container_tab_indicator.dart]: same typo

File: lib/container_tab_indicator.dart (hunk 1 above)
Comment (slovnicki) [reply lib/container_tab_indicator.dart:83]: huh, I'll have to take a closer look at this later in the night. I don't remember much here

Comment (jhb-dev) [reply lib/container_tab_indicator.dart:83]: Okay 👍

**Comment by slovnicki** [comment]
I didn't have time to look at this, will do during the week.

**Comment by jhb-dev** [comment]
> I didn't have time to look at this, will do during the week.

Yeah, no stress! :)

File: lib/container_tab_indicator.dart (hunk 1 above)
Comment (slovnicki) [reply lib/container_tab_indicator.dart:83]: I think this is fine

**Review by slovnicki — APPROVED** [review]
LGTM :slightly_smiling_face:

**Pull request merged**